### PR TITLE
Reverted Google auth functionality for legacy use

### PR DESCRIPTION
**Pull request by EvanBacon**
Because we are not focusing on adding Firebase authentication, we should revert this module to use the old Google API console.
Google auth being a part of Firebase was important for using firebase invites, which is now deprecated 🙃
This revert will account for users working in Expo vs unmanaged/standalone. There will be some repetitive functionality but considering how many issues have been opened, I think it'll be fine.

# Test Plan

**`androidStandaloneAppClientId` & `iosStandaloneAppClientId`**

Using in ejected Expo should have the same functionality as a standalone app. ie: Bundle ID & Package will be custom and should be registered by the app.json before ejecting.

**`androidClientId` & `iosClientId`**

Create a new project in Google API (not Firebase) and use `host.exp.Exponent` as the bundle ID and package.

### CODE REVIEW

**Comment by EvanBacon** [comment]
Once I get approval on the docs, I'll retroactively apply them.

**Comment by EvanBacon** [comment]
Assumes this is merged: #3402

File: packages/expo/src/Google/Google.ts [hunk 1]
```diff
@@ -25,53 +41,123 @@ export type LogInResult =
...
+
+  return { isValid: true };
+}
+
+function guidFromClientId(clientId: string): string {
+  const clientIdComponents = clientId.split('.').filter(component => component.includes('-'));
+
+  const guid = clientIdComponents[0];
+  const isValid = isValidGUID(guid);
+  if (!isValid.isValid) {
```
Comment (ide) [inline packages/expo/src/Google/Google.ts]: lol name this a bit differently:
```suggestion
  if (!validation.isValid) {
```

File: packages/expo/src/Google/Google.ts [hunk 2]
```diff
@@ -25,53 +41,123 @@ export type LogInResult =
...
+  if (components.length !== 2) {
+    return {
+      isValid: false,
+      reason: `\`${guid}\` must be a string of numbers and an alphanumeric string ${PROJECT_ID_LENGTH} characters long, joined with a hyphen.`,
+    };
+  }
+  const projectNumber = components[0];
+  const projectId = components[1];
+  if (isNaN(+projectNumber)) {
+    const hashedProjectId = Array(PROJECT_ID_LENGTH).map((l, i) => (i % 2 === 0 ? 'X' : 'x'));
```
Comment (ide) [inline packages/expo/src/Google/Google.ts]: `Array(PROJECT_ID_LENGTH).fill('x')` seems simpler unless you want the XxXxX for a specific reason

File: packages/expo/src/Google/Google.ts (hunk 1 above)
Comment (EvanBacon) [reply packages/expo/src/Google/Google.ts]: woopsies

File: packages/expo/src/Google/Google.ts [hunk 3]
```diff
@@ -25,53 +41,121 @@ export type LogInResult =
...
+    }) || clientId;
+
+  if (
+    typeof iosClientId === 'string' &&
+    typeof androidClientId === 'string' &&
+    iosClientId === androidClientId
+  ) {
+    throw new CodedError(
+      'ERR_GOOGLE_CONFIG',
+      'Keys for Android an iOS cannot be the same value. Ensure you are linking the client IDs matching the given platforms in the Google APIs console: https://console.developers.google.com/apis/credentials'
```
Comment (ide) [inline packages/expo/src/Google/Google.ts]:
```suggestion
      'Keys for Android and iOS cannot be the same value. Ensure you are linking the client IDs matching the given platforms in the Google APIs console: https://console.developers.google.com/apis/credentials'
```

File: packages/expo/src/Google/Google.ts [hunk 4]
```diff
@@ -1,13 +1,29 @@
-import { AppAuth } from 'expo-app-auth';
-import { UnavailabilityError } from 'expo-errors';
+import * as AppAuth from 'expo-app-auth';
+import { CodedError } from 'expo-errors';
 import { Platform } from 'react-native';
+import Constants from 'expo-constants';
 
+const isInExpo = Constants.appOwnership === 'expo';
 export type LogInConfig = {
```
Comment (ide) [inline packages/expo/src/Google/Google.ts]:
```suggestion
export type GoogleLogInConfig = {
```

**Review by ide — APPROVED** [review]

**Comment by mkyo2012** [comment]
problem solved?

**Pull request merged**